Tests/SwiftDocCTests/Infrastructure/ExternalReferenceResolverTests.swift

```diff
@@ -1155,4 +1155,232 @@ class ExternalReferenceResolverTests: XCTestCase {
+        )
+    }
+
+    func withExampleDocumentation(_ files: [any File], path: String, block: (Symbol) throws -> Void) throws {
+        let exampleDocumentation = Folder(name: "unit-test.docc", content: files)
+
+        let resolver = TestExternalReferenceResolver()
+
+        let tempURL = try createTempFolder(content: [exampleDocumentation])
+        let (_, bundle, context) = try loadBundle(from: tempURL, externalResolvers: [resolver.bundleIdentifier: resolver])
+        XCTAssert(context.problems.isEmpty, "Unexpected problems:\n\(context.problems.map(\.diagnostic.summary).joined(separator: "\n"))")
+
+        let reference = ResolvedTopicReference(bundleIdentifier: bundle.identifier, path: path, sourceLanguage: .swift)
+        let node = try context.entity(with: reference)
+        let symbol = try XCTUnwrap(node.semantic as? Symbol)
+        try block(symbol)
+    }
+
+    func withExampleSymbolDocumentation(_ files: File..., block: (Symbol) throws -> Void) throws {
+        let symbolGraph = JSONFile(
+            name: "ModuleName.symbols.json",
+            content: makeSymbolGraph(
+                moduleName: "ModuleName",
+                symbols: [
+                    SymbolGraph.Symbol(
+                        identifier: .init(precise: "symbol-id", interfaceLanguage: "swift"),
+                        names: .init(title: "SymbolName", navigator: nil, subHeading: nil, prose: nil),
+                        pathComponents: ["SymbolName"],
+                        docComment: nil,
+                        accessLevel: .public,
+                        kind: .init(parsedIdentifier: .class, displayName: "Kind Display Name"),
+                        mixins: [:]
+                    )
+                ]
+            )
+        )
+        try withExampleDocumentation(
+            [symbolGraph] + files,
+            path: "/documentation/ModuleName/SymbolName",
+            block: block
+        )
+    }
+
+    func testParametersWithExternalLink() throws {
+        try withExampleSymbolDocumentation(
+            TextFile(name: "Extension.md", utf8Content: """
+            # ``SymbolName``
+
+            This is about some symbol.
+
+            - Parameters:
+              - one: The first parameter has a link: <doc://com.external.testbundle/something/related/to/this/param>.
+              - two: The second parameter also has a link: <doc://com.external.testbundle/something/related/to/this/param>.
+
+            """)
+        ) { symbol in
+            let expectedMarkupsWithResolvedLink = [
+                "The first parameter has a link: <doc://com.external.testbundle/externally/resolved/path>.",
+                "The second parameter also has a link: <doc://com.external.testbundle/externally/resolved/path>."
+            ]
+            let parametersSection = try XCTUnwrap(symbol.parametersSection)
+            XCTAssertEqual(parametersSection.parameters.count, 2)
+            try parametersSection.parameters.forEach {
```

Comment:
nit: I don't feel that `forEach` is any more readable than a regular `for-in` loop here (actually, pretty much never but that's a different topic 😉 ).

Reply:
Ha ha another Ruby idiom. In Ruby we almost always write `some_collection.each do ... end` with a block/closure, and almost never use a for loop, although it is supported. Again the idea is to make the Ruby code a bit more readable.


But since this is Swift I've refactored things to use for-loops :)